src/shazamkit.cs

```diff
@@ -0,0 +1,253 @@
+		[Field ("SHMediaItemISRC")]
+		Isrc,
+		[Field ("SHMediaItemMatchOffset")]
+		MatchOffset,
+		[Field ("SHMediaItemFrequencySkew")]
+		FrequencySkew,
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHCatalog
+	{
+		[Export ("minimumQuerySignatureDuration")]
+		double MinimumQuerySignatureDuration { get; }
+
+		[Export ("maximumQuerySignatureDuration")]
+		double MaximumQuerySignatureDuration { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHCatalog))]
+	interface SHCustomCatalog
+	{
+		[Export ("addReferenceSignature:representingMediaItems:error:")]
+		bool Add (SHSignature signature, SHMediaItem[] mediaItems, [NullAllowed] out NSError error);
+
+		[Export ("addCustomCatalogFromURL:error:")]
+		bool Add (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("writeToURL:error:")]
+		bool Write (NSUrl url, [NullAllowed] out NSError error);
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[Export ("new")]
+		[return: Release]
+		SHCustomCatalog Create ();
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMatch : NSSecureCoding
+	{
+		[Export ("mediaItems", ArgumentSemantic.Strong)]
+		SHMatchedMediaItem[] MediaItems { get; }
+
+		[Export ("querySignature", ArgumentSemantic.Strong)]
+		SHSignature QuerySignature { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHMediaItem))]
+	[DisableDefaultCtor]
+	interface SHMatchedMediaItem : NSSecureCoding
+	{
+		[Export ("frequencySkew")]
+		float FrequencySkew { get; }
+
+		[Export ("matchOffset")]
+		double MatchOffset { get; }
+
+		[Export ("predictedCurrentMatchOffset")]
+		double PredictedCurrentMatchOffset { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaItem : NSSecureCoding, NSCopying
+	{
+		[NullAllowed, Export ("shazamID")]
+		string ShazamId { get; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; }
+
+		[NullAllowed, Export ("subtitle")]
+		string Subtitle { get; }
+
+		[NullAllowed, Export ("artist")]
+		string Artist { get; }
+
+		[Export ("genres", ArgumentSemantic.Strong)]
+		string[] Genres { get; }
+
+		[NullAllowed, Export ("appleMusicID")]
+		string AppleMusicId { get; }
+
+		[NullAllowed, Export ("appleMusicURL", ArgumentSemantic.Strong)]
+		NSUrl AppleMusicUrl { get; }
+
+		[NullAllowed, Export ("webURL", ArgumentSemantic.Strong)]
+		NSUrl WebUrl { get; }
+
+		[NullAllowed, Export ("artworkURL", ArgumentSemantic.Strong)]
+		NSUrl ArtworkUrl { get; }
+
+		[NullAllowed, Export ("videoURL", ArgumentSemantic.Strong)]
+		NSUrl VideoUrl { get; }
+
+		[Export ("explicitContent")]
+		bool ExplicitContent { get; }
+
+		[NullAllowed, Export ("isrc")]
```

Comment:
Possible, but I think @spouliot had a point in leaving the export alone rather than combining it with other attributes.

Reply:
```suggestion
		[NullAllowed] [Export ("isrc")]
```
Do you mean doing something like this?

Reply:
Do you mean doing something like ```[NullAllowed][Export ("isrc")]```?

Reply:
Like all the other bindings, one line for just the export.